model/src/main/kotlin/OrtResult.kt

```diff
@@ -234,16 +234,17 @@ data class OrtResult(
      */
     fun collectProjectsAndPackages(includeSubProjects: Boolean = true): SortedSet<Identifier> {
         val projectsAndPackages = sortedSetOf<Identifier>()
-        val allSubProjects = sortedSetOf<Identifier>()
+
+        getProjects().mapTo(projectsAndPackages) { it.id }
 
         if (!includeSubProjects) {
+            val allSubProjects = sortedSetOf<Identifier>()
+
             getProjects().forEach {
                 it.collectSubProjects().mapTo(allSubProjects) { ref -> ref.id }
             }
-        }
 
-        getProjects().mapNotNullTo(projectsAndPackages) { project ->
-            project.id.takeUnless { it in allSubProjects }
+            projectsAndPackages -= allSubProjects
         }
 
         getPackages().mapTo(projectsAndPackages) { it.pkg.id }
```

Comment:
Or the other way around, drop the more recently introduced `getProjectAndPackageIds` again?

FYI, IIRC the rationale for calling these functions "colllect..." was to avoid the need for `@JsonIgnore` for them.

Reply:
> Or the other way around, drop the more recently introduced `getProjectAndPackageIds` again?

yeah sure, whatever make more sense.

Reply:
> if you agree we could just make a PR renaming those to get*, I'd could take on this task if you want.

I'd prefer if we don't do this, but instead introduce a new file named e.g. `OrtResultExtensions.kt`, and move (almost?) all current OrtResult member function as extension functions to that new file. In there, it should be possible to rename functions to "get..." without the need for `@JsonIgnore`. And we should consolidate duplicate functions in the same run. And add missing documentation.

@fviernau, if that doesn't collide with any of your work I can take that task.

Reply:
I cannot see any problem with ```@JsonIgnore``` so I don't get why one would convert the functions to extension functions. In particular since this eliminates the possibility for using ```private / member``` variables and thus adds complexity without need and removes some ```optimization potential```. Can you explain why you want to avoid ```JsonIgnore``` ?

Anyhow to answer you question, doing so does not collide with any of my work.

Reply:
I regard `@JsonIgnore` as clutter that makes the code harder to read (I feel so for all annotations BTW; I strongly prefer language built-ins over annotation, if possible).

Also it has happened more than once in the past, that we forgot to add `@JsonIgnore` to functions in this class, so I regard this approach as fragile.

@mnonnenmacher, any opinion on member functions vs. extension functions here?

Reply:
@sschuberth what do you think about the shortcomings of the extensions approach I outlined above? I think those a valid concerns and should not be ```ingored``` in the discussion.

Even if the current approach was fragile, I think we have more additional options than just ```member```vs.  ```extension```.

Reply:
> I think those a valid concerns and should not be `ingored` in the discussion.

They are valid concerns and I don't intend to ignore them. Obviously, I don't want to simply exchange fragile / hard to maintain code with code that is hard to maintain just in another way. But I haven't looked closely enough to even make up my mind whether this is the case yet.